lib/LANraragi/Plugin/Metadata/ComicInfo.pm

```diff
@@ -0,0 +1,118 @@
+package LANraragi::Plugin::Metadata::ComicInfo;
+
+use strict;
+use warnings;
+
+use Mojo::DOM;
+
+#You can also use the LRR Internal API when fitting.
+use LANraragi::Model::Plugins;
+use LANraragi::Utils::Logging qw(get_plugin_logger);
+use LANraragi::Utils::Archive qw(is_file_in_archive extract_file_from_archive);
+
+#Meta-information about your plugin.
+sub plugin_info {
+
+    return (
+        #Standard metadata
+        name         => "ComicInfo",
+        type         => "metadata",
+        namespace    => "comicinfo",
+        author       => "Gin-no-kami",
+        version      => "1.0",
+        description  => "Parses metadata from ComicInfo.xml embedded in the archive",
+        parameters => []
+    );
+
+}
+
+#Mandatory function to be implemented by your plugin
+sub get_tags {
+
+    shift;
+    my $lrr_info = shift; # Global info hash, contains various metadata provided by LRR
+
+    #Use the logger to output status - they'll be passed to a specialized logfile and written to STDOUT.
+    my $logger = get_plugin_logger();
+
+    my $file = $lrr_info->{file_path};
+    my $path_in_archive = is_file_in_archive( $file, "ComicInfo.xml" );
+
+    if ( $path_in_archive ) {
+        #Extract ComicInfo.xml
+        my $filepath = extract_file_from_archive( $file, $path_in_archive );
+
+        #Read file into string
+        my $stringxml = "";
+        open( my $fh, '<:encoding(UTF-8)', $filepath )
+          or return ( error => "Could not open $filepath!" );
+        while ( my $line = <$fh> ) {
+            chomp $line;
+            $stringxml .= $line;
+        }
+
+        #Parse file into DOM object and extract tags
+        my $genre;
+        my $group;
+        my $url;
+        my $artist;
+        my $lang;
+        my $result = Mojo::DOM->new->xml(1)->parse($stringxml)->at('Genre');
```

Comment:
You could technically only do the XML parsing once and store the DOM object to reuse later:  

```
my $dom = Mojo::DOM->new->xml(1)->parse($stringxml);

if (defined $dom->at('Genre')) {
  $genre = $dom->at('Genre')->text;
}
# etc
```  

Not a big deal though, fine to merge without that fixed.